bin/sozo/src/commands/options/signer.rs

```diff
@@ -0,0 +1,237 @@
+use std::str::FromStr;
+
+use anyhow::{anyhow, Result};
+use clap::Args;
+use dojo_world::metadata::Environment;
+use starknet::core::types::FieldElement;
+use starknet::signers::{LocalWallet, SigningKey};
+
+use super::{DOJO_KEYSTORE_PASSWORD_ENV_VAR, DOJO_KEYSTORE_PATH_ENV_VAR, DOJO_PRIVATE_KEY_ENV_VAR};
+
+#[derive(Debug, Args)]
+#[command(next_help_heading = "Signer options")]
+// INVARIANT:
+// - For commandline: we can either specify `private_key` or `keystore_path` along with
+//   `keystore_password`. This is enforced by Clap.
+// - For `Scarb.toml`: if both private_key and keystore are specified in `Scarb.toml` private_key
+//   will take priority
+pub struct SignerOptions {
+    #[arg(long, env = DOJO_PRIVATE_KEY_ENV_VAR)]
+    #[arg(conflicts_with = "keystore_path")]
+    #[arg(help_heading = "Signer options - RAW")]
+    #[arg(help = "The raw private key associated with the account contract.")]
+    pub private_key: Option<String>,
+
+    #[arg(long = "keystore", env = DOJO_KEYSTORE_PATH_ENV_VAR)]
+    #[arg(value_name = "PATH")]
+    #[arg(help_heading = "Signer options - KEYSTORE")]
+    #[arg(help = "Use the keystore in the given folder or file.")]
+    pub keystore_path: Option<String>,
+
+    #[arg(long = "password", env = DOJO_KEYSTORE_PASSWORD_ENV_VAR)]
+    #[arg(value_name = "PASSWORD")]
+    #[arg(help_heading = "Signer options - KEYSTORE")]
+    #[arg(help = "The keystore password. Used with --keystore.")]
+    pub keystore_password: Option<String>,
+}
+
+impl SignerOptions {
+    pub fn signer(&self, env_metadata: Option<&Environment>) -> Result<LocalWallet> {
+        if let Some(private_key) =
+            self.private_key.as_deref().or_else(|| env_metadata.and_then(|env| env.private_key()))
+        {
+            return Ok(LocalWallet::from_signing_key(SigningKey::from_secret_scalar(
+                FieldElement::from_str(private_key)?,
+            )));
+        }
+
+        if let Some(path) = &self
+            .keystore_path
+            .as_deref()
+            .or_else(|| env_metadata.and_then(|env| env.keystore_path()))
+        {
+            if let Some(password) = self
+                .keystore_password
+                .as_deref()
+                .or_else(|| env_metadata.and_then(|env| env.keystore_password()))
+            {
+                return Ok(LocalWallet::from_signing_key(SigningKey::from_keystore(
+                    path, password,
+                )?));
+            } else {
+                return Err(anyhow!("Keystore path is specified but password is not."));
```

Comment:
Now that we support the keystore, we should ask for the user to input the password in this situation (exactly as it's done in `inspect-private` for instance).

Reply:
[Prompt password instead of error](https://github.com/dojoengine/dojo/pull/1601/commits/44f4cb21289cf9143487cb2766adf839d4e21731)

Reply:
Great, now we can have the `keystore_path` into the `Scarb.toml` and then it prompts when needed.